pkg/common/outcomes/types.go

```diff
@@ -2,15 +2,12 @@ package outcomes
 
 // Outcome is the printable context returned from a check (common.Runnable).
 type Outcome interface {
-	// GetShortStatus returns a short status of the check outcome, such as success, fail, or diagnostic.
-	GetShortStatus() string
+	// GetOutcomeType returns the type/status of the check outcome: pass/fail/info/unknown.
+	GetOutcomeType() string
 
-	// GetLongDiagnostics returns detailed diagnostics that were dynamically-generated during the check.
-	// Certain checks (such as checking whether the pod participates in an SMI policy)
-	// cannot be easily categorized as successful or failed but instead as diagnostic checks.
-	// These diagnostic checks require dynamically-generated diagnostic info to be
-	// shown to the user to give them more context.
-	GetLongDiagnostics() string
+	// GetDiagnostics returns detailed diagnostics that were dynamically-generated during the check.
+	// Diagnostics may include details about a test failure or information about the mesh/its components
```

Comment:
Are diagnostics used for a failed outcome?

Reply:
You are very correct! hm we should address this and GetError() as well yes

Reply:
Having some trouble removing this because it's used by printable outcomes. Let's deal with this in a follow up PR